src/main/java/sg/ncl/MainController.java

```diff
@@ -1186,7 +1188,33 @@ public String checkApplyTeamInfo(
         HttpEntity<String> request = createHttpEntityWithBody(mainObject.toString());
         ResponseEntity response = restTemplate.exchange(properties.getRegisterRequestToApplyTeam(session.getAttribute("id").toString()), HttpMethod.POST, request, String.class);
 
-        return "redirect:/teams/team_application_submitted";
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+
+                if (error.getName().equals(ExceptionState.ApplyNewProjectException.toString())) {
+                    logger.info("Apply new team fail at adapter deterlab");
+                    redirectAttributes.addFlashAttribute("message", error.getMessage());
+                } else if (error.getName().equals(ExceptionState.RegisterTeamNameDuplicateException.toString())) {
+                    logger.info("Apply new team fail: team name already exists", teamPageApplyTeamForm.getTeamName());
+                    redirectAttributes.addFlashAttribute("message", "Team name already exists.");
+                } else {
+                    logger.info("Apply new team fail: registration service or adapter fail");
+                    // possible sio or adapter connection fail
+                    redirectAttributes.addFlashAttribute("message", ERR_SERVER_OVERLOAD);
+                }
+                return "redirect:/teams/apply_team";
+
+            } else {
+                // no errors, everything ok
+                logger.info("Completed invoking the apply team request service for Team: {}", teamPageApplyTeamForm.getTeamName());
+                return "redirect:/teams/team_application_submitted";
+            }
+        } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)